api/lib/domain/services/certification-center-invitation-service.js

```diff
@@ -0,0 +1,55 @@
+import * as maillingService from './mail-service.js';
+import { CertificationCenterInvitation } from '../models/index.js';
+import {
+  SendingEmailError,
+  SendingEmailToInvalidDomainError,
+  SendingEmailToInvalidEmailAddressError,
+} from '../errors.js';
+
+const createOrUpdateCertificationCenterInvitation = function ({
+  certificationCenterInvitationRepository,
+  mailService = maillingService,
+}) {
+  return async function ({ certificationCenter, email, locale }) {
+    let certificationCenterInvitation =
+      await certificationCenterInvitationRepository.findOnePendingByEmailAndCertificationCenterId({
+        certificationCenterId: certificationCenter.id,
+        email,
+      });
+
+    if (!certificationCenterInvitation) {
+      const certificationCenterInvitationToCreate = CertificationCenterInvitation.create({
+        certificationCenterId: certificationCenter.id,
+        email,
+      });
+
+      certificationCenterInvitation = await certificationCenterInvitationRepository.create(
+        certificationCenterInvitationToCreate,
+      );
+    }
+
+    const mailerResponse = await mailService.sendCertificationCenterInvitationEmail({
+      certificationCenterInvitationId: certificationCenterInvitation.id,
+      certificationCenterName: certificationCenter.name,
+      code: certificationCenterInvitation.code,
+      email,
+      locale,
+    });
+
+    if (mailerResponse?.status === 'FAILURE') {
```

Comment:
Je suggère plutôt la forme ci-dessous : 

```suggestion
    if (mailerResponse.status !== 'SUCCESS') {
```

En effet le `?.` est inutile dans ce cas. Cela aurait été utile sir on avait eu un cas comme `mailerResponse.meta?.status` (c'est un exemple imaginaire, il n'y a pas de `meta` dans `mailerResponse`)
cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

De plus, même si cela revient au même compte tenu du code du mailService, il est plus sûr de tester si on a bien un succès plutôt que de tester si on a une erreur.